### PR TITLE
Feature/tg 10 login

### PR DESCRIPTION
**Pull request by Delebrith**

### CODE REVIEW

**Comment by Grushenko** [comment]
What about refresh token (with corespondong endpoit) for JWT authorization?
I'd prefer to sepate authorization from user CRUD requests by placing auth endpoits in path like: 
* `/auth/authorize` for authorization and token retrieval
* `/auth/refresh` for exchanging refresh token for auth token
* ? `/auth/me` for user data retreival (could be merged with responses from two previous endpoints or could be placed in user endpoint branch e.g. `/user/me` (but that creates confilct with standard REST API patern `/user/{userID}`). It seems the best option is to have it in `/auth/me` cause its generally connected with user personal data (authenticity)

**Comment by Grushenko** [comment]
* `auth/refresh` should also generate new refresh token and return it with authToken
* `auth/refresh` represents exchanging refreshToken for new authToken and refreshToken. As it is not indempotent operation it should be implemented as `POST` request
* best prcatice is to make refrshToken 'one-use only' - when exchanging refresh token for new authToken new refreshToken should be generated and the one used in that request should be invaliated. Also it is not required for refresh to have form of JWT token, securitywise it's good to be able to revoke refreshTokens in case of dataleaks and stuff. 
* its better to use `401 Unauthorized` responses for accessing resources when not authenticated and  use `403 Forbidden` for cases when user is authenticated but does not have permissions for accessing that resource. That gives clear information to client if it shoud try to refresh authToken (`401`) or not (`403`).

File: apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/authentication/AuthTokenType.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+import java.time.Duration;
+
+@RequiredArgsConstructor
+@Getter
+public enum AuthTokenType {
+    ACCESS_TOKEN(Duration.ofHours(2)),
```
Comment (Grushenko) [inline apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/authentication/AuthTokenType.java:11]: Duration of access token could be lower (e.g. 10-15 min), but in our system, where we do not use tokens to store access rights (roles, permisions etc.) that can change at any time, 2 hours should be fine.

Comment (Delebrith) [reply apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/authentication/AuthTokenType.java:11]: We can adjust it and add removing tokens (by a scheduled job, maybe?) later in development. If everything else if fine, let's merge the PR so we can start working on other user stories

**Pull request merged**